x/ccv/provider/keeper/proposal_test.go

```diff
@@ -845,69 +853,88 @@ func TestBeginBlockCCR(t *testing.T) {
-		{ChainId: "2", SpawnTime: now.Add(2 * time.Hour)},
 		{ChainId: "3", SpawnTime: now.Add(3 * time.Hour)},
+		{ChainId: "1", SpawnTime: now.Add(1 * time.Hour)},
 		{ChainId: "4", SpawnTime: now.Add(4 * time.Hour)},
+		{ChainId: "2", SpawnTime: now.Add(2 * time.Hour)},
 	}
 
 	keeperParams := testkeeper.NewInMemKeeperParams(t)
 	providerKeeper, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, keeperParams)
 	defer ctrl.Finish()
 
 	for _, prop := range props {
-		cpProp := prop // bring into loop scope - avoids using iterator pointer instead of value pointer
-		err := providerKeeper.SetPendingConsumerAdditionProp(ctx, &cpProp)
-		require.NoError(t, err)
+		require.NotPanics(t, func() {
+			cpProp := prop // bring into loop scope - avoids using iterator pointer instead of value pointer
+			providerKeeper.SetPendingConsumerAdditionProp(ctx, &cpProp)
+		})
 	}
 
 	// advance the clock to be 1 minute after first proposal
 	ctx = ctx.WithBlockTime(now.Add(time.Minute))
-	res := providerKeeper.GetAllConsumerAdditionProps(ctx)
-	require.NotEmpty(t, res, "GetAllConsumerAdditionProps returned empty result")
-	require.Len(t, res.Pending, 4, "wrong len for pending addition props")
-	require.Equal(t, props[0].ChainId, res.Pending[0].ChainId, "wrong chain ID for pending addition prop")
+	storedProps := types.ConsumerAdditionProposals{}
+	providerKeeper.IteratePendingConsumerAdditionProps(ctx, func(prop types.ConsumerAdditionProposal) (stop bool) {
+		storedProps.Pending = append(storedProps.Pending, &prop)
+		return false // do not stop the iteration
+	})
+	require.NotEmpty(t, storedProps)
+	require.Len(t, storedProps.Pending, 4, "wrong len for pending addition props")
+	// check that the order is correct
+	require.Equal(t, props[1].ChainId, storedProps.Pending[0].ChainId)
+	require.Equal(t, props[3].ChainId, storedProps.Pending[1].ChainId)
+	require.Equal(t, props[0].ChainId, storedProps.Pending[2].ChainId)
+	require.Equal(t, props[2].ChainId, storedProps.Pending[3].ChainId)
 }
 
-// Test getting both matured and pending consumer removal proposals
-func TestGetAllConsumerRemovalProps(t *testing.T) {
+// Test iterating through all consumer removal proposals
+func TestIteratePendingConsumerRemovalProps(t *testing.T) {
 	now := time.Now().UTC()
 
 	props := []types.ConsumerRemovalProposal{
-		{ChainId: "1", StopTime: now.Add(1 * time.Hour)},
-		{ChainId: "2", StopTime: now.Add(2 * time.Hour)},
 		{ChainId: "3", StopTime: now.Add(3 * time.Hour)},
+		{ChainId: "1", StopTime: now.Add(1 * time.Hour)},
 		{ChainId: "4", StopTime: now.Add(4 * time.Hour)},
+		{ChainId: "2", StopTime: now.Add(2 * time.Hour)},
 	}
 
 	keeperParams := testkeeper.NewInMemKeeperParams(t)
 	providerKeeper, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, keeperParams)
 	defer ctrl.Finish()
 
 	for _, prop := range props {
-		providerKeeper.SetPendingConsumerRemovalProp(ctx, prop.ChainId, prop.StopTime)
+		require.NotPanics(t, func() {
```

Comment:
Curious why this `NotPanics` explicit call is needed? Tests fail upon panic right